### PR TITLE
Load user-defined deployment settings

### LINKED ISSUE #4579: Deployment profile loader doesn't load user-defined object-like settings
### Actual Behavior

The profile loader throws away all the fields in the `WSL.integrations`, `diagnostics.mutedChecks`, and `extensions` objects.

The loader wants to see a schema for these values, but there isn't one because they're user-defined.

I'm writing a unit test that will fail on these right now.

### PR DESCRIPTION
**Pull request by ericpromislow**
Fixes #4579

Don't filter out default settings for values like `WSL.integrations`, `diagnostics.mutedChecks`, or `extensions`.

### CODE REVIEW

File: pkg/rancher-desktop/main/deploymentProfiles.ts [hunk 1]
```diff
@@ -258,31 +261,46 @@ function readRegistryUsingSchema(schemaObj: any, regKey: nativeReg.HKEY): Recurs
  * Do simple type validation of a deployment profile
  * @param profile The profile to be validated
  * @param schema The structure (usually defaultSettings) used as a template
+ * @param parentPathParts The parent path for the current schema key.
  * @returns The original profile, less any invalid fields
  */
-function validateDeploymentProfile(profile: any, schema: any) {
+function validateDeploymentProfile(profile: any, schema: any, parentPathParts: string[]) {
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply pkg/rancher-desktop/main/deploymentProfiles.ts:267]: Yeah, that simplifies the code. I prefer `if...else if... else if ... else ...` all at the same level when possible, and more or less got there.

File: pkg/rancher-desktop/main/deploymentProfiles.ts [hunk 2]
```diff
@@ -258,31 +261,46 @@ function readRegistryUsingSchema(schemaObj: any, regKey: nativeReg.HKEY): Recurs
...
+        console.log(`Deployment Profile ignoring '${ parentPathParts.join('.') }.${ key }': not in schema.`);
         delete profile[key];
       }
     }
   }
 
   return profile;
 }
+
+function isSpecialObject(pathParts: string[], key: string): boolean {
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply pkg/rancher-desktop/main/deploymentProfiles.ts:296]: I also gave it a better name.

File: pkg/rancher-desktop/main/deploymentProfiles.ts [hunk 3]
```diff
@@ -258,31 +261,46 @@ function readRegistryUsingSchema(schemaObj: any, regKey: nativeReg.HKEY): Recurs
...
-              console.log(`Deployment Profile ignoring '${ key }'. Array type mismatch.`);
+        if (typeof profile[key] === 'object') {
+          if (Array.isArray(profile[key])) {
+            if (!Array.isArray(schema[key])) {
+              console.log(`Deployment Profile ignoring '${ parentPathParts.join('.') }.${ key }': got an array, expecting type ${ typeof schema[key] }.`);
               delete profile[key];
-            } else if (!Array.isArray(profile[key])) {
-              validateDeploymentProfile(profile[key], schema[key]);
             }
+          } else if (parentPathParts.length <= 2 && isSpecialObject(parentPathParts, key)) {
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply pkg/rancher-desktop/main/deploymentProfiles.ts:277]: Sure.

File: pkg/rancher-desktop/main/deploymentProfiles.ts [hunk 4]
```diff
@@ -141,17 +141,20 @@ async function convertAndParsePlist(inputPath: string): Promise<undefined|Recurs
   try {
     plutilResult = await spawnFile('plutil', args, { stdio: [body, 'pipe', 'pipe'] });
   } catch (error: any) {
+    // throw error if plutil fails to parse the 'locked' profile
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply pkg/rancher-desktop/main/deploymentProfiles.ts:144]: Probably snuck in during a rebase

File: pkg/rancher-desktop/main/deploymentProfiles.ts [hunk 5]
```diff
@@ -141,17 +141,20 @@ async function convertAndParsePlist(inputPath: string): Promise<undefined|Recurs
...
     console.log(`Error parsing deployment profile plist file ${ inputPath }\n${ error }`);
     throw new DeploymentProfileError(`Error loading plist file ${ inputPath }: ${ getErrorString(error) }`);
   }
 
   try {
     return JSON.parse(plutilResult.stdout ?? '');
   } catch (error: any) {
+    // throw error if we fail to parse the 'locked' profile
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply pkg/rancher-desktop/main/deploymentProfiles.ts:152]: same

File: pkg/rancher-desktop/main/deploymentProfiles.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (adamkpickering) [reply pkg/rancher-desktop/main/deploymentProfiles.ts:265]: It's a lot easier to read now. Thanks!!

**Review by adamkpickering — APPROVED** [review]

**Pull request merged**